var/spack/repos/builtin/packages/acts/package.py

```diff
@@ -228,10 +234,29 @@ class Acts(CMakePackage, CudaPackage):
         description="Build python bindings for the examples",
         when="@14: +examples",
     )
+    variant(
+        "svg",
+        default=False,
+        description="Build ActSVG display plugin",
+        when="@20.1:",
+    )
+    variant(
+        "tbb",
+        default=True,
+        description="Build the examples with Threading Building Blocks library",
+        when="@19.8:19 +examples",
+    )
+    variant(
+        "tbb",
+        default=True,
+        description="Build the examples with Threading Building Blocks library",
+        when="@20.1: +examples",
+    )
```

Comment:
Can't this be merged into one variant with `when="@19.8:19,20.1: +examples"`?

Reply:
Indeed, this doesn't work. Even though the multiple variant when conditions are represented like that in `spack info`. E.g. `spack info acts` has the line
```console
Variants:
    Name [Default]                 When                              Allowed values          Description
    ===========================    ==============================    ====================    =========================================================

    examples [off]                 [@17:+fatras+identification+js    on, off                 Build the examples
                                   on+tgeo, @0.23:16+digitization
                                   +fatras+identification+json+tg
                                   eo]
```
for the two defined variants for `examples`.

Following this representation logically, it would probably need to be `when="@19.8:19 +examples, @20.1: +examples"`. But this fails. Even the simpler `when="@19.8:19, @20.1:"` fails with
```console
==> [2022-10-16-15:45:45.149817] SpecParseError: Invalid version specifier
==> [2022-10-16-15:45:45.149877] Error: Invalid version specifier
  Encountered when parsing spec:
    @19.8:19,@20.1:
             ^
Traceback (most recent call last):
  File "/home/wdconinc/git/spack/lib/spack/spack/spec.py", line 5164, in do_parse
    specs.append(self.spec(None))
  File "/home/wdconinc/git/spack/lib/spack/spack/spec.py", line 5269, in spec
    vlist = self.version_list()
  File "/home/wdconinc/git/spack/lib/spack/spack/spec.py", line 5364, in version_list
    vlist.append(self.version())
  File "/home/wdconinc/git/spack/lib/spack/spack/spec.py", line 5352, in version
    self.next_token_error("Invalid version specifier")
  File "/home/wdconinc/git/spack/lib/spack/spack/parse.py", line 129, in next_token_error
    raise ParseError(message, self.text[0], self.token.end)
spack.parse.ParseError: Invalid version specifier
```

I cannot find an issue about this. I think it's a pretty major change, though, since it adds an operator to the spec grammar and it needs to behave 'as expected'. Case in point: should it be `@19.8:19,20.1: +examples` (operator has high precedence and only works inside versions), or `@19.8:19,@20.1: +examples` (operator has lower precedence), or `@19.8:19 +examples, @20.1: +examples` (lowest precedence)?

Reply:
Hmm, so it does seem to work when using the highest precedence comma operator... but only if the other variants are shared as they are in this case.